MAINTAINERS.md

```diff
@@ -0,0 +1,50 @@
+# How to release a new version
+
+## When to release a new version
+
+New version should be released when we are ready to make changes generally available.
+
+New version should not be released if we want to test our latest changes or to make them available to a limited number of users. This can be achieved without releasing a new version. Go modules allow to point to latest master or specific Git commit.
+
+Consider releasing a new Release Candidate version to make changes available to a larger group of users, if we are not ready to make them available to everyone yet.
+
+Under no circumstances, official releases must be done during the weekend or during the wee hours of the night.
+
+## Version numbers
+
+We follow the rules for semantic versioning, but prefix with the letter `v`.
+
+Examples of official releases:
+- `v1.0.0`
+- `v1.0.3`
+- `v1.2.3`
+- `v2.1.7`
+
+Examples of Release Candidates:
+- `v1.0.0-rc.4`
+- `v2.1.0-rc.1`
+
+## Step-by-step release process
+
+1. Clone or fetch latest master of these Git repositories:
```

Comment:
Should we note here that this isn't the ideal end state? It isn't usually necessary for an upstream library to attempt to checkout all its downstream users and try the new code with them.

Reply:
This document reflects the current state of the process. When we change the process, we should update the document as well.